src/cljs/rems/administration/resource.cljs

```diff
@@ -1,97 +1,133 @@
 (ns rems.administration.resource
   (:require [clojure.string :as str]
             [re-frame.core :as rf]
+            [rems.administration.components :refer [text-field]]
             [rems.autocomplete :as autocomplete]
             [rems.collapsible :as collapsible]
             [rems.text :refer [text localize-item]]
             [rems.util :refer [dispatch! fetch post!]]))
 
-(defn- fetch-licenses []
-  (fetch "/api/licenses?active=true"
-         {:handler #(do (rf/dispatch [::set-licenses %])
-                        (rf/dispatch [::set-selected-licenses #{}]))}))
-
-(defn- create-resource [prefix resid licenses]
-  (post! "/api/resources/create" {:params {:prefix prefix
-                                           :resid resid
-                                           :licenses (if licenses
-                                                       (map :id licenses)
-                                                       [])}
+(defn- valid-request? [request]
+  (and (not (str/blank? (:prefix request)))
+       (not (str/blank? (:resid request)))))
+
+(defn- build-request [form]
+  (let [request {:prefix (:prefix form)
+                 :resid (:resid form)
+                 :licenses (map :id (:licenses form))}]
+    (when (valid-request? request)
+      request)))
+
+(defn- create-resource [form]
+  (post! "/api/resources/create" {:params (build-request form)
```

Comment:
what if build-request returns nil?

Reply:
The check for nil happens already when the save button is pressed. Though maybe it would be cleaner to send the request object instead of the form object to create-resource.